File: src/main/java/com/examportal/entities/Answer.java
package com.examportal.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Entity
@Table(name = "answers")
@NoArgsConstructor
@Getter
@Setter
@ToString(callSuper = true)
public class Answer extends BaseEntity {
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "exam_id", nullable = false)
    private Exam exam;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "question_id", nullable = false)
    private Question question;
    
    @Column(length = 1)
    private String selectedOption;
    
    @Column(nullable = false)
    private Boolean isCorrect;
    
    @Column(nullable = false)
    private Integer score;

    public Answer(User user, Exam exam, Question question, String selectedOption, 
                 Boolean isCorrect, Integer score) {
        this.user = user;
        this.exam = exam;
        this.question = question;
        this.selectedOption = selectedOption;
        this.isCorrect = isCorrect;
        this.score = score;
    }
}

File: Dotnet3/ExamPortal/Models/Answer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ExamPortal.Models
{
    public class Answer : BaseEntity
    {
        [Required]
        public long QuestionId { get; set; }
        
        [Required]
        public long UserId { get; set; }
        
        [Required]
        [MaxLength(1)]
        public string SelectedOption { get; set; } = string.Empty;
        
        // Navigation properties
        [ForeignKey("QuestionId")]
        public virtual Question Question { get; set; } = null!;
        
        [ForeignKey("UserId")]
        public virtual User User { get; set; } = null!;
    }
}
